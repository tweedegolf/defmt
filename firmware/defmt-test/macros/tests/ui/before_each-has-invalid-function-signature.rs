fn main() {}

#[defmt_test_macros::tests]
mod tests {
    #[before_each]
    fn hello(a: i32, b: i32) -> i32 {
        a + b
    }
}
