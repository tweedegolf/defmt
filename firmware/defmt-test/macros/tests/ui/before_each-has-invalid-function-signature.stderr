error: `#[before_each]` function must have signature `fn(state: &mut Type)` (parameter is optional)
 --> tests/ui/before_each-has-invalid-function-signature.rs:6:8
  |
6 |     fn hello(a: i32, b: i32) -> i32 {
  |        ^^^^^
